debuglogger/default.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package debuglogger
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"strconv"
+	"strings"
+)
+
+type defaultEvent struct {
+	level  LogLevel
+	logger *log.Logger
+	fields []byte
+}
+
+func (e *defaultEvent) Msg(msg string) {
+	if len(msg) == 0 {
+		return
+	}
+
+	s := strings.Builder{}
+	s.WriteString("[")
+	s.WriteString(e.level.String())
+	s.WriteString("] ")
+	s.WriteString(msg)
+	s.Write(e.fields)
+
+	e.logger.Println(s.String())
+}
+
+func (e *defaultEvent) Str(key, val string) Event {
+	e.fields = append(e.fields, ' ')
+	e.fields = append(e.fields, key...)
+	e.fields = append(e.fields, '=')
+	e.fields = append(e.fields, val...)
+	return e
+}
+
+func (e *defaultEvent) Err(err error) Event {
+	if err == nil {
+		return e
+	}
+
+	e.fields = append(e.fields, " error=\""...)
+	e.fields = append(e.fields, err.Error()...)
+	e.fields = append(e.fields, '"')
+	return e
+}
+
+func (e *defaultEvent) Bool(key string, b bool) Event {
+	e.fields = append(e.fields, ' ')
+	e.fields = append(e.fields, key...)
+	e.fields = append(e.fields, '=')
+	if b {
+		e.fields = append(e.fields, "true"...)
+	} else {
+		e.fields = append(e.fields, "false"...)
+	}
+	return e
+}
+
+func (e *defaultEvent) Int(key string, i int) Event {
+	e.fields = append(e.fields, ' ')
+	e.fields = append(e.fields, key...)
+	e.fields = append(e.fields, '=')
+	e.fields = append(e.fields, strconv.Itoa(i)...)
+	return e
+}
+
+func (e *defaultEvent) Uint(key string, i uint) Event {
+	e.fields = append(e.fields, ' ')
+	e.fields = append(e.fields, key...)
+	e.fields = append(e.fields, '=')
+	e.fields = append(e.fields, strconv.Itoa(int(i))...)
+	return e
+}
+
+func (e *defaultEvent) Stringer(key string, val fmt.Stringer) Event {
+	e.fields = append(e.fields, ' ')
+	e.fields = append(e.fields, key...)
+	e.fields = append(e.fields, '=')
+	e.fields = append(e.fields, val.String()...)
+	return e
+}
+
+type defaultLogger struct {
+	*log.Logger
+	level LogLevel
+}
+
+func (l defaultLogger) WithOutput(w io.Writer) Logger {
+	if l.Logger == nil {
+		return defaultLogger{
+			Logger: log.New(w, "", log.LstdFlags),
+			level:  l.level,
+		}
+	}
+
+	return defaultLogger{
+		Logger: log.New(w, l.Logger.Prefix(), l.Logger.Flags()),
+		level:  l.level,
+	}
+}
+
+func (l defaultLogger) WithLevel(lvl LogLevel) Logger {
+	return defaultLogger{Logger: l.Logger, level: lvl}
+}
+
+func (l defaultLogger) Trace() Event {
+	if l.level < LogLevelTrace || l.Logger == nil {
```

Comment:
Don't think `Logger` can be `nil`, especially after making it a private field as it probably should be

Reply:
It was in NOP case but I just changed it based on this convenience https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/log/log.go;l=72